apiserver/resources_mig.go

```diff
@@ -156,9 +153,17 @@ func queryToResource(query url.Values) (charmresource.Resource, error) {
 	if err != nil {
 		return empty, errors.BadRequestf("invalid size")
 	}
-	res.Fingerprint, err = charmresource.ParseFingerprint(query.Get("fingerprint"))
-	if err != nil {
-		return empty, errors.BadRequestf("invalid fingerprint")
+	switch res.Type {
+	case charmresource.TypeFile:
+		if res.Path == "" {
+			return empty, errors.BadRequestf("missing path")
+		}
+		res.Fingerprint, err = charmresource.ParseFingerprint(query.Get("fingerprint"))
+		if err != nil {
+			return empty, errors.BadRequestf("invalid fingerprint")
+		}
+	case charmresource.TypeContainerImage:
+		res.Fingerprint = charmresource.Fingerprint{}
```

Comment:
Won't this be the value by default? Are you just doing this to be explicit?

Reply:
(That's obviously admirable, just making sure I understand what's happening.)

Reply:
yeah, just to be explicit